test/e2e/actions/cloud/provider.go

```diff
@@ -0,0 +1,358 @@
+	ID         string
+	Region     string
+	Targets    []string
+	SubnetName string
+}
+
+func (r *GCPPrivateEndpointRequest) isPrivateEndpointRequest() {}
+
+type PrivateEndpointDetails struct {
+	ProviderName      provider.ProviderName
+	Region            string
+	ID                string
+	IP                string
+	GCPProjectID      string
+	EndpointGroupName string
+	Endpoints         []GCPPrivateEndpoint
+}
+
+type GCPPrivateEndpoint struct {
+	Name string
+	IP   string
+}
+
+func (a *ProviderAction) SetupPrivateEndpoint(request PrivateEndpointRequest) *PrivateEndpointDetails {
+	a.t.Helper()
+
+	switch req := request.(type) {
+	case *AWSPrivateEndpointRequest:
+		ID, err := a.awsProvider.CreatePrivateEndpoint(req.ServiceName, req.ID, req.Region)
+		Expect(err).To(BeNil())
+
+		return &PrivateEndpointDetails{
+			ProviderName: provider.ProviderAWS,
+			Region:       req.Region,
+			ID:           ID,
+		}
+	case *GCPPrivateEndpointRequest:
+		endpoints := make([]GCPPrivateEndpoint, 0, len(req.Targets))
+		for index, target := range req.Targets {
+			rule, ip, err := a.gcpProvider.CreatePrivateEndpoint(req.ID, req.Region, req.SubnetName, target, index)
+			Expect(err).To(BeNil())
+
+			endpoints = append(
+				endpoints,
+				GCPPrivateEndpoint{
+					Name: rule,
+					IP:   ip,
+				},
+			)
+		}
+
+		return &PrivateEndpointDetails{
+			ProviderName:      provider.ProviderGCP,
+			Region:            req.Region,
+			GCPProjectID:      a.gcpProvider.projectID,
+			EndpointGroupName: a.gcpProvider.network.VPC,
+			Endpoints:         endpoints,
+		}
+	case *AzurePrivateEndpointRequest:
+		pe, err := a.azureProvider.CreatePrivateEndpoint(vpcName, req.SubnetName, req.ID, req.ServiceResourceID, req.Region)
+		Expect(err).To(BeNil())
+		Expect(pe).ShouldNot(BeNil())
+		Expect(pe.Properties).ShouldNot(BeNil())
+		Expect(pe.Properties.NetworkInterfaces).ShouldNot(BeNil())
+		Expect(pe.Properties.NetworkInterfaces).ShouldNot(BeEmpty())
+
+		var itf *armnetwork.Interface
+		Eventually(func(g Gomega) bool {
+			itf, err = a.azureProvider.GetInterface(path.Base(*pe.Properties.NetworkInterfaces[0].ID))
+			g.Expect(err).To(BeNil())
+			g.Expect(itf).ShouldNot(BeNil())
+			g.Expect(itf.Properties).ShouldNot(BeNil())
+			g.Expect(itf.Properties.IPConfigurations).ShouldNot(BeNil())
+			g.Expect(itf.Properties.IPConfigurations).ShouldNot(BeEmpty())
+			g.Expect(itf.Properties.IPConfigurations[0].Properties).ShouldNot(BeNil())
+
+			return true
+		}).WithTimeout(5 * time.Minute).WithPolling(15 * time.Second).Should(BeTrue())
+
+		return &PrivateEndpointDetails{
+			ProviderName: provider.ProviderAzure,
+			Region:       req.Region,
+			ID:           *pe.ID,
+			IP:           *itf.Properties.IPConfigurations[0].Properties.PrivateIPAddress,
+		}
+	}
+
+	return nil
+}
+
+func (a *ProviderAction) ValidatePrivateEndpointStatus(providerName provider.ProviderName, endpoint, region string, gcpNumAttachments int) {
+	a.t.Helper()
+
+	Eventually(func() bool {
```

Comment:
Shouldn't this function take the `g Gomega` argument?

Reply:
good catch!